streams/src/test/java/org/apache/kafka/streams/integration/utils/EmbeddedKafkaCluster.java

```diff
@@ -118,8 +120,25 @@ private void putIfAbsent(final Properties props, final String propertyKey, final
      * Stop the Kafka cluster.
      */
     private void stop() {
+        if (brokers.length > 1) {
+            // delete the topics first to avoid cascading leader elections while shutting down the brokers
+            final Set<String> topics = getAllTopicsInCluster();
```

Comment:
I thought this could throw, too?

Reply:
I guess it could, but it's just directly reading the topics from zookeeper (of which there's only one node in the cluster). If that fails, it seems like everything else would also fail, so it's probably pointless to even bother closing the brokers. Similarly, calling `shutdown` on the first broker could also throw, but we don't bother catching there to continue calling brokers on the rest of the cluster.

Reply:
Ok. Works for me.